File: Report-Analyzer/src/backend/database/models/testMethod.ts
import { Schema } from 'mongoose';
import { IScoreMap, ScoreMapSchmaObject, ScoreSchemaObject } from './score';


export interface ITestMethod {
  Description: string,
  TestDescription: string,
  TlsVersion: string,
  RFC: {
    Section: string,
    Number: number
  }
  MethodName: string,
  DisplayName: string,
  ClassName: string
}

export const TestMethodSchemaObject = {
  Description: String,
  TestDescription: String,
  TlsVersion: String,
  RFC: {
    Section: String,
    Number: Number
  },
  MethodName: String,
  DisplayName: String,
  ClassName: String
}

File: Report-Analyzer/src/backend/uploader/src/database/models/testMethod.go
package models

type TestMethod struct {
	Description     string `bson:"Description"`
	TestDescription string `bson:"TestDescription"`
	TlsVersion      string `bson:"TlsVersion"`
	RFC             struct {
		Section string `bson:"Section"`
		Number  int    `bson:"Number"`
	} `bson:"RFC"`
	MethodName  string `bson:"MethodName"`
	DisplayName string `bson:"DisplayName"`
	ClassName   string `bson:"ClassName"`
}
